### ISSUE TITLE
[BUG] Unknown error while attempting to download a module from a repo

### ISSUE THREAD

**Issue by lucker999** [MRepoApp/MRepo#176]
[###] Checklist

- [X] No one has submitted a similar or identical bug report before
- [X] I'm using the latest version of MRepo

### Bug description

In Non-Root mode I attempt to download modules from tow repo (IzzyOnDroid and Magisk Modules Alt Repo (mirror)) it doesn't work: instantly get a notification with a message "Unknown error". BTW, the same behavior occurs if I change the mode to Root (Magisk).

### Expected behavior

Module downloaded once I tap on the button "Download"

### Actual behavior

Instantly get a notification with a message "Unknown error", no download happens.

### Steps to reproduce

1. Install the latest version of Mrepo (currently 2.4.4 (713)) 
2. Add repos IzzyOnDroid and Magisk Modules Alt Repo (mirror)
3. Open a module card and choose the tab "Version"
4. Select the latest version of a module and tap on the button "Download" at the bottom of the screen
5. Get notification with a message "Unknown error", no download happens

### UI / OS

crDroid 10.2 (release from 2024.02.10)

### Android Version

14

### Additional info

OnePlus 11
Can manually download modules from the same repo in a browser and successfully install them, which is great but hugely inconvenient. Hope you can fix this.
![Screenshot_20240307-173750_MRepo](https://github.com/MRepoApp/MRepo/assets/9690776/c48c73cc-ba57-498d-a7e2-633f0ca2d9a2)

**Comment by SanmerDev** [comment]
Can you provide logs?

**Comment by lucker999** [comment]
This is form an app MatLog (crDroid native), but I'm not sure if this is enough. If you provide some guidance on how to accuire a log I would be much obliged since I'm just a user (somewhat advanced), but not an engineer :)

**Comment by SanmerDev** [comment]
You can try changing the download path, but I'm not sure this will work.

**Comment by lucker999** [comment]
Already done hat. Started with defaults (Downloads), then created a directory Magisk and changed to it. No changes.

**Comment by SanmerDev** [comment]
@lucker999, I added a button to export the logs in top bar of Settings page.

**Comment by lucker999** [comment]
Thank you very much for the efforts, I appreciate your nice gesture. I installed the release by the link you kindly built and, surprisingly, it works flawlessly. I successfully tested downloads in the default directory (Downloads), but changing it appears to be somewhat tricky: for some reason I cannot choose a directory I created in the root of internal storage. Not that it pose any problem. I created a directory (Magisk) in the Downloads and that works too.
Encouraged by that, I freshly installed the generic Mrepo release from F-Droid and it works too. So, wrapping this up, we just spent some time chasing non-existing bug. I'm sorry about that, though have no idea why this has happened. Mrepo was one of half of dozens the very first apps I installed on the new phon with freshly flashed latest crDroid. 
Once again, thank you for the effort and hope you will develop this awesome tool further.
